config.go

```diff
@@ -0,0 +1,236 @@
+	}
+	desthost := ""
+	pflag.StringVarP(&desthost, "dest-host", "d", "", "Destination syslog hostname or IP")
+	if desthost != "" {
+		self.DestHost = desthost
+	}
+	destport := 0
+	pflag.IntVarP(&destport, "dest-port", "p", 0, "Destination syslog port")
+	if destport != 0 {
+		self.DestPort = destport
+	}
+	foreground := true
+	if utils.CanDaemonize {
+		pflag.BoolVarP(&foreground, "no-detach", "D", false, "Don't daemonize and detach from the terminal")
+	}
+	self.Daemonize = !foreground
+	// facility
+	var s string
+	pflag.StringVarP(&s, "facility", "f", "user", "Facility")
+	facility, err := syslog.Facility(s)
+	if err != nil {
+		return fmt.Errorf("%s is not a designated facility", s)
+	}
+	self.Facility = facility
+	pflag.StringVar(&self.Hostname, "hostname", "", "Local hostname to send from")
+	pflag.StringVar(&self.PidFile, "pid-file", "", "Location of the PID file")
+	// severity
+	pflag.StringVarP(&s, "severity", "s", "notice", "Severity")
+	severity, err := syslog.Severity(s)
+	if err != nil {
+		return fmt.Errorf("Invalid severity: %s", s)
+	}
+	self.Severity = severity
+	// --strip-color
+	pflag.BoolVar(&self.UseTCP, "tcp", false, "Connect via TCP (no TLS)")
+	pflag.BoolVar(&self.UseTLS, "tls", false, "Connect via TCP with TLS")
+	pflag.BoolVar(&self.Poll, "poll", false, "Detect changes by polling instead of inotify")
+	pflag.StringVar(&s, "new-file-check-interval", "", "How often to check for new files")
+	if s != "" {
+		if err := self.RefreshInterval.Set(s); err != nil {
+			return err
+		}
+	}
+	_ = pflag.Bool("no-eventmachine-tail", false, "No action, provided for backwards compatibility")
+	_ = pflag.Bool("eventmachine-tail", false, "No action, provided for backwards compatibility")
+	logfile := ""
+	pflag.StringVar(&logfile, "debug-log-cfg", "", "the debug log file")
+	if logfile != "" {
+		self.DebugLogFile = logfile
+	}
+	pflag.StringVar(&self.LogLevels, "log", "<root>=INFO", "\"logging configuration <root>=INFO;first=TRACE\"")
+	pflag.Parse()
+	self.Files = append(self.Files, pflag.Args()...)
+	return nil
+}
+
+func (self *Config) validate() error {
+	// hostname
+	if self.Hostname == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			return fmt.Errorf("Could not obtain host name: %v", err)
+		}
+		self.Hostname = hostname
+	}
+	// destination host
+	if self.DestHost == "" {
+		return fmt.Errorf("No destination hostname specified")
+	}
+	// destination port
+	if self.DestPort == 0 {
+		self.DestPort = 514
+	}
+	// protocol
+	if self.Protocol == "" {
+		switch {
+		case self.UseTLS:
+			self.Protocol = "tls"
+		case self.UseTCP:
+			self.Protocol = "tcp"
+		default:
+			self.Protocol = "udp"
+		}
+	}
+	// root CAs
+	if self.Protocol == "tls" &&
+		self.DestHost == "logs.papertrailapp.com" {
+		self.RootCAs = papertrail.RootCA()
+	}
+	// log file
+	if self.DebugLogFile == "" {
+		self.DebugLogFile = "/dev/null"
+	}
+	// refresh interval
+	if self.RefreshInterval == 0 {
+		self.RefreshInterval = MIN_REFRESH_INTERVAL
```

Comment:
@troy are you happy for this to default to 10 seconds but allow a minimum of 1 second?